src/services/mongoose/users.ts

```diff
@@ -5,6 +5,8 @@ import {
   RepositoryError,
 } from "@domain/errors"
 import { User } from "@services/mongoose/schema"
+import { caseInsensitiveRegex } from "@core/utils"
```

Comment:
This function is only used by functions that should be part of the `UsersRepository`. I would move it into `mongoose/users.ts` file and out of utils.